packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -35,6 +37,14 @@ const datePartRegex = new RegExp(/^\s*([0-9]{4})-([0-9]{1,2})-([0-9]{1,2})\s*$/)
  */
 const timePartRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
 
+function isDateValid(date) {
+	if (date === undefined) {
+		return true;
+	}
+
+	return date instanceof Date && !isNaN(date.getTime());
```

Comment:
```javascript
const date = new Date("I'm the best language to have bug because I don't throw errors if developers do s*** x)");
date === undefined; //    > false 
date instanceof Date; //  > true
date.getTime() //         > NaN 
```

Reply:
✔